brave-spancollector-scribe/src/test/java/com/github/kristofa/brave/scribe/ScribeSpanCollectorTest.java

```diff
@@ -93,7 +82,39 @@ public void testCollectWithDefaultAnnotation() throws TTransportException {
         final BinaryAnnotation binaryAnnotation = binary_annotations.get(0);
         assertEquals(KEY1, binaryAnnotation.getKey());
         assertEquals(VALUE1, new String(binaryAnnotation.getValue()));
+    }
 
+    @Test
+    public void testNotifyOfSpanCollectionEvents() throws TTransportException {
+        // given
+        final CountingScribeSpanCollectorNotificationsHandler eventsHandler = new CountingScribeSpanCollectorNotificationsHandler();
+        ScribeSpanCollectorParams params = new ScribeSpanCollectorParams();
+        params.setNotificationsHandler(eventsHandler);
+
+        final ScribeSpanCollector zipkinSpanCollector = new ScribeSpanCollector("localhost", PORT, params);
+
+        // when
+        collectSingleSpan(zipkinSpanCollector);
+
+        // then
+        assertEquals(0, eventsHandler.getLastQueueSize());
+        assertEquals(1, eventsHandler.getTotalSpansSentEvents());
+        assertEquals(0, eventsHandler.getTotalSendingFailures());
+        assertEquals(1, eventsHandler.getLastBatchSize());
+        assertEquals(0, eventsHandler.getTotalSpansRejected());
+    }
+
+    private void collectSingleSpan(ScribeSpanCollector zipkinSpanCollector) {
```

Comment:
I'd probably remove this method and inline it in the test, particularly as you can use try-with-resources on ScribeSpanCollector